File: ToyMachineInterpreterConsole/src/Model/Expression/ConstantExpression.java
package Model.Expression;

import Model.Utils.IDictionary;
import Model.Utils.IHeap;

public class ConstantExpression implements IExpression {
    private int value;

    public ConstantExpression(int value) {
        this.value = value;
    }

    @Override
    public int eval(IDictionary<String, Integer> dict, IHeap<Integer, Integer> heap) {
        return value;
    }

    public String toString() {
        return "" + value;
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Expression/ConstantExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.Containers;

namespace Toy_Language_CS_Interpreter.Model.Expression
{
    class ConstantExpression:IExpression
    {
        private int value;

        public ConstantExpression(int value)
        {
            this.value = value;
        }

        public int Evaluate(Containers.IDictionary<string, int> symbolTable)
        {
            return value;
        }

        public override string ToString()
        {
            return "" + value;
        }
    }
}
